### ISSUE TITLE
[llvm] FileNotFoundError if C++ compiler is not installed

### ISSUE THREAD

**Issue by ChrisCummins** [facebookresearch/CompilerGym#360]
## 🐛 Bug

When the LLVM environment is instantiated without a working C++ compiler on the system, a `FileNotFoundError` is raised.

## To Reproduce

Steps to reproduce the behavior using Docker:

1. Create a `Dockerfile`:
```
FROM python:3.8-slim-buster
RUN pip3 install compiler_gym
CMD [ "python3", "-m", "compiler_gym.bin.service", "--env=llvm-v0"]
```
2. Build and run it to trigger the error:
```
$ docker build -t test . && docker run test

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "www.py", line 239, in describe
    with compiler_gym.make("llvm-v0") as env:
  File "/usr/local/lib/python3.8/site-packages/compiler_gym/util/registration.py", line 16, in make
    return gym.make(id, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/gym/envs/registration.py", line 184, in make
    return registry.make(id, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/gym/envs/registration.py", line 106, in make
    env = spec.make(**kwargs)
  File "/usr/local/lib/python3.8/site-packages/gym/envs/registration.py", line 76, in make
    env = cls(**_kwargs)
  File "/usr/local/lib/python3.8/site-packages/compiler_gym/envs/llvm/llvm_env.py", line 108, in __init__
    datasets=_get_llvm_datasets(site_data_base=datasets_site_path),
  File "/usr/local/lib/python3.8/site-packages/compiler_gym/envs/llvm/llvm_env.py", line 70, in _get_llvm_datasets
    _LLVM_DATASETS = list(get_llvm_datasets(site_data_base=site_data_base))
  File "/usr/local/lib/python3.8/site-packages/compiler_gym/envs/llvm/datasets/__init__.py", line 253, in get_llvm_datasets
    yield CsmithDataset(site_data_base=site_data_base, sort_order=0)
  File "/usr/local/lib/python3.8/site-packages/compiler_gym/envs/llvm/datasets/csmith.py", line 139, in __init__
    self.clang_compile_command: List[str] = ClangInvocation.from_c_file(
  File "/usr/local/lib/python3.8/site-packages/compiler_gym/envs/llvm/llvm_benchmark.py", line 132, in command
    for directory in get_system_includes():
  File "/usr/local/lib/python3.8/site-packages/compiler_gym/envs/llvm/llvm_benchmark.py", line 106, in get_system_includes
    _SYSTEM_INCLUDES = list(_get_system_includes())
  File "/usr/local/lib/python3.8/site-packages/compiler_gym/envs/llvm/llvm_benchmark.py", line 44, in _get_system_includes
    process = subprocess.Popen(
  File "/usr/local/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/local/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'c++'
```

## Expected behavior

The LLVM environment should still be usable without a system compiler. Perhaps just generate a warning.

## Environment

Please fill in this checklist:

- CompilerGym: 0.1.9
- How you installed CompilerGym (conda, pip, source): pip
- OS: Linux
- Python version: 3.8
